internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriterFactory.java

```diff
@@ -39,16 +38,19 @@ public class VeniceWriterFactory {
   private final String defaultBrokerAddress;
 
   public VeniceWriterFactory(Properties properties) {
-    this(properties, PubSubClientsFactory.createProducerFactory(properties), null);
+    this(properties, null);
   }
 
   public VeniceWriterFactory(Properties properties, MetricsRepository metricsRepository) {
-    this(properties, PubSubClientsFactory.createProducerFactory(properties), metricsRepository);
+    this(
+        properties,
+        PubSubClientsFactory.fromVeniceProperties(new VeniceProperties(properties)).getProducerAdapterFactory(),
```

Comment:
We can simply pass null and centralize the factory creation logic in the main constructor.